axon/src/lib.rs

```diff
@@ -212,13 +209,26 @@ pub fn rlp_opt_list<T: Encodable>(rlp: &mut RlpStream, opt: &Option<T>) {
 }
 
 pub fn convert_client_id_to_string(client_id: [u8; 32]) -> String {
-    let s = format!("{:x}", H256::from(client_id));
-    s
+    format!("{:x}", H256::from(client_id))
 }
 
-pub fn convert_string_to_client_id(s: &str) -> [u8; 32] {
-    let a = H256::from_str(s).unwrap().into();
-    a
+pub fn convert_string_to_client_id(s: &str) -> Result<[u8; 32], VerifyError> {
+    Ok(H256::from_str(s)
+        .map_err(|_| VerifyError::WrongClient)?
+        .into())
+}
+
+pub fn convert_connection_id_to_index(connection_id: &str) -> Result<usize, VerifyError> {
+    let index_str = connection_id
+        .split('-')
```

Comment:
Where is the defined rule of `connection_id`?

Reply:
it's defined just in the code, I leave a comment above that function to explain it, besides, I'm working on the coding of the adoption of packet, this comment will be written after new commits